src/Package/Impl/DataInspect/VariableViewModel.cs

```diff
@@ -61,7 +61,7 @@ public sealed class VariableViewModel : RSessionDataObject, IIndexedItem {
                 ShowDetailCommandTooltip = Resources.ShowDetailCommandTooltip;
             }
 
-            CanShowOpenCsv = result.CanCoerceToDataFrame && result.Length > 1;
+            CanShowOpenCsv = result.CanCoerceToDataFrame && (result.Length > 1 || (result.Length == 1 && result.TypeName == "S4"));
```

Comment:
No need to do the `==1` check for S4 - they are length 1 by default, but can override to mean something else, and there's no reason to suppress export for those that do override length (but provide `as.data.frame`).